python/main_loop.py

```diff
@@ -125,20 +143,28 @@ def createNewLocalPath(sailbot, maxAllowableRuntimeSeconds, desiredHeadingPublis
         isGlobalWaypointReached = (localPath.waypointReached(state.position, previousGlobalWaypoint, nextGlobalWaypoint,
                                                              True) or isLastWaypointReached)
 
+        if(lowWindConditions):
+            # ensure we havent updated local path recently before permitting
+            lowWindPermitsUpdate = (createNewLocalPathAttempts >= LOW_WIND_LOOPS_PER_PATH_UPDATE)
+            createNewLocalPathAttempts += 1
+        else:
+            lowWindPermitsUpdate = True
+
         newGlobalPathReceived = sailbot.newGlobalPathReceived
         reachedEndOfLocalPath = localPath.reachedEnd()
         pathNotReachGoal = not localPath.reachesGoalLatlon(state.globalWaypoint)
-        mustUpdateLocalPath = (hasUpwindOrDownwindOnPath or hasObstacleOnPath or isGlobalWaypointReached
+        mustUpdateLocalPath = ((hasUpwindOrDownwindOnPath and lowWindPermitsUpdate)
+                               or hasObstacleOnPath or isGlobalWaypointReached
                                or newGlobalPathReceived or reachedEndOfLocalPath or pathNotReachGoal
                                or localPathUpdateForced)
```

Comment:
The idea here is that we only update the local path if the current path is "invalid" for reasons that actually matter in light wind. For example, I think we want to make sure the path is updated normally when we reach a global waypoint so that we at least have a plan for where we want to go in the future. However, upwind/downwind sailing are meaningless in light wind, so this is not a "valid" reason to update the local path.

Reply:
Makes sense!